src/main/java/org/zalando/nakadi/validation/schema/SchemaGeneration.java

```diff
@@ -0,0 +1,66 @@
+package org.zalando.nakadi.validation.schema;
+
+import com.google.common.collect.Lists;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.skyscreamer.jsonassert.JSONCompare;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+
+public class SchemaGeneration {
+    public JSONObject mergeSchema(final JSONObject thisSchema, final JSONObject thatSchema) {
+        if (thisSchema.length() == 0) {
+            return thatSchema;
+        // TODO: ugly hack to generate anyOf schemas. Could be optimized to generate smaller schemas.
+        } else if (JSONCompare.compareJSON(thisSchema, thatSchema, JSONCompareMode.STRICT).failed()) {
+            return new JSONObject().put("anyOf", new JSONArray(Lists.newArrayList(thisSchema, thatSchema)));
+        } else {
+            return thisSchema;
+        }
+    }
+
+    public JSONObject schemaFor(final Object event) {
+        if (event instanceof String) {
+            return new JSONObject().put("type", "string");
+        } else if (event instanceof Integer) {
+            return new JSONObject().put("type", "number");
+        } else if (event instanceof Boolean) {
+            return new JSONObject().put("type", "boolean");
+        } else if (event == JSONObject.NULL) {
+            return new JSONObject().put("type", "null");
+        } else if (event instanceof JSONArray) {
+            return schemaFor((JSONArray) event);
```

Comment:
Yes 👍 if I removed this line of code, it would break this test case here https://github.com/zalando/nakadi/pull/519/files#diff-e0ec26c222e2d4ccfcfd792d8838ef00R96 for example.